### PR TITLE
Proper splitting of arguments for mixin node

### PR DESCRIPTION
**Pull request by dusanmsk**
I have fast-fixed issue https://github.com/neuland/jade4j/issues/90 for me, so only sharing a code if you like it.

### CODE REVIEW

File: src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private static final char argumentDelimiter = ',';
+    private final String arguments;
+    private List<String> argList = new ArrayList<String>();
+    private StringBuffer sb;
+
+    public static List<String> split(String arguments) {
+        return new ArgumentSplitter(arguments).splitArguments();
+    }
+
+    public ArgumentSplitter(String arguments) {
```
Comment (marlonbernardes) [inline src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java]: There is no point exposing this constructor if there is not even a single public method.

File: src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package de.neuland.jade4j.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author dusan.zatkovsky, 2/5/15
+ */
+public class ArgumentSplitter {
```
Comment (marlonbernardes) [inline src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java]: You should add some tests to this class.

File: src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    public static List<String> split(String arguments) {
+        return new ArgumentSplitter(arguments).splitArguments();
+    }
+
+    public ArgumentSplitter(String arguments) {
+        this.arguments = arguments;
+    }
+
+    private List<String> splitArguments() {
```
Comment (marlonbernardes) [inline src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java]: This method could be improved in so many ways: 
- StringBuilder is always preffered over a StringBuffer 
- You don't really need to create a private attribute for StringBuffer - it's only appending one element to the arguments list before being discard (you are reinstantiating it every time `pushArg` is called.
- The method itself is confusing. I could help you to refactor it, but I'm afraid to do that without some tests.

File: src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java (hunk 2 above)
Comment (dusanmsk) [inline src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java]: Well done

File: src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java (hunk 3 above)
Comment (dusanmsk) [inline src/main/java/de/neuland/jade4j/util/ArgumentSplitter.java]: Refactored a little bit, StringBuilder now as local member.

**Comment by naltatis** [comment]
Looks good. Could you remove the `pom.xml` change from the PR?

**Comment by dusanmsk** [comment]
My colleague found another usecase where my argument splitter fail too, so please hold the merge, he is trying to fix it.

**Comment by dusanmsk** [comment]
Fixed (with very basic braces support) + tests.

**Comment by dusanmsk** [comment]
bump on this. thx.

**Comment by dusanmsk** [comment]
Bump again. When this will be accepted? I would like to drop my own build of jade4j ...

**Comment by FinalGuy** [comment]
Hi dusanmsk,

i'll have a look at it now. Thank you for your contribution.

**Comment by dusanmsk** [comment]
Thank you, perfect.

**Pull request merged**